src/util/locking/RedisResourceLocker.ts

```diff
@@ -0,0 +1,113 @@
+import type { RedisClient } from 'redis';
+import { createClient } from 'redis';
+import type { Lock } from 'redlock';
+import Redlock from 'redlock';
+import type { ResourceIdentifier } from '../../ldp/representation/ResourceIdentifier';
+import { getLoggerFor } from '../../logging/LogUtil';
+import { InternalServerError } from '../errors/InternalServerError';
+import type { ResourceLocker } from './ResourceLocker';
+
+/**
+ * A locking system that uses a Redis server or any number of
+ * Redis nodes / clusters
+ */
+export class RedisResourceLocker implements ResourceLocker {
+  protected readonly logger = getLoggerFor(this);
+
+  protected readonly clients: RedisClient[];
+  public readonly redlock: Redlock;
+
+  private readonly lockList: Map<string, Lock>;
```

Comment:
@RubenVerborgh This is going to cause this class to not work in a multithreaded environment. So while this class would be an alternative to our current locking implementation, it doesn't actually solve the issue of having a ResourceLocker that works in a multithreaded environment. This is a problem caused by the library being used though, and I had a quick look but can't actually see any npm Redis locking library that would not have this issue.

Reply:
That's insane—that would be the whole point of using Redis. To be investigated.

Reply:
The only thing that happens here if locally saving the lock to later release it(wether the lock itself is active or not is saved on the Redis server). This isn't done in any Redlock demo or tutorial but has to be done to work with `acquire()` and `release()`.

A second instance of the server should not be able to acquire a lock on a resource if the first instance has that lock.
As far as I understand, this is what the Redlock library does for us.
(I hope I am not completely wrong on this one)

Reply:
True, that is already better than what we have now (note that the `acquire` function should not error in that case, but simply resolve once it can acquire the lock).

But the issue is that a second instance can not release a lock acquired by the first instance, which is required for a read/write locking algorithm.

Reply:
@joachimvh, can you explain why we would want a process to be able to release locks aquired by another process? Isn't the entire point of the acquire–release locking mechanism that the acquiring process waits for the lock to be released by the process that locked it?

Reply:
Actually the interface was chosen specifically so one process could do the locking and another could do the unlocking (which is not possible with the `withReadLock` interface for example). The reason for this is read/write locking algorithms that allow multiple simultaneous read operations that share a single lock for this, as is done in [GreedyReadWriteLocker](https://github.com/solid/community-server/blob/594cf27696a7c8a59231b8e3e593089816459811/src/util/locking/GreedyReadWriteLocker.ts) for example.